### PR TITLE
Issue #533 - i18n.

### PR DESCRIPTION
**Pull request by phillip-kruger**
Also replaced slf4j with JUL in the API module(as this needs to boil up to the spec, so we should not introduce new dependency)

For all other modules replaced slf4J with JBoss Logger and adding i18n.

I had to remove annotationprocessor in the compile plugin in the pom, but the build still works.

For some tests (like mosqito) I had to keep slf4j, but for most replace with JBoss Logger.
Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -59,7 +59,6 @@
   <properties>
     <vertx.version>3.9.1</vertx.version>
     <rxjava.version>2.2.19</rxjava.version>
-    <slf4j-api.version>1.7.30</slf4j-api.version>
```
Comment (cescoffier) [inline pom.xml:62]: I believe the documentation must be updated as it refers SL4JF.

File: smallrye-reactive-messaging-amqp/pom.xml [hunk 2]
```diff
@@ -62,14 +57,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-amqp/pom.xml:65]: Why did you remove this?

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package io.smallrye.reactive.messaging.amqp.i18n;
+
+import org.jboss.logging.Messages;
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageBundle;
+
+@MessageBundle(projectCode = "SRRML")
```
Comment (cescoffier) [inline smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:8]: What does the code mean? SmallRye Reactive Messaging L? Logging? Logger? Library?

**Comment by kenfinnigan** [comment]
Already conflicts ;-)

File: smallrye-reactive-messaging-aws-sns/pom.xml [hunk 4]
```diff
@@ -82,14 +77,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-aws-sns/pom.xml:85]: This must be kept.

File: smallrye-reactive-messaging-camel/pom.xml [hunk 5]
```diff
@@ -65,14 +65,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-camel/pom.xml:68]: This must be kept.

File: smallrye-reactive-messaging-http/pom.xml [hunk 6]
```diff
@@ -67,14 +67,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-http/pom.xml:70]: Same, must be kept.

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java [hunk 7]
```diff
@@ -137,15 +143,15 @@ public static void clear() {
     public PublisherBuilder<? extends Message<?>> getPublisherBuilder(Config config) {
         String name = config.getOptionalValue("channel-name", String.class)
                 .orElseThrow(
-                        () -> new IllegalArgumentException("Invalid incoming configuration, `channel-name` is not set"));
+                        () -> ex.illegalArgumentInvalidIncomingConfig());
```
Comment (cescoffier) [inline smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java:146]: Can it use a method reference?

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java [hunk 8]
```diff
@@ -137,15 +143,15 @@ public static void clear() {
...
         return sources.computeIfAbsent(name, InMemorySourceImpl::new).source;
     }
 
     @Override
     public SubscriberBuilder<? extends Message<?>, Void> getSubscriberBuilder(Config config) {
         String name = config.getOptionalValue("channel-name", String.class)
                 .orElseThrow(
-                        () -> new IllegalArgumentException("Invalid outgoing configuration, `channel-name` is not set"));
+                        () -> ex.illegalArgumentInvalidOutgoingConfig());
```
Comment (cescoffier) [inline smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java:154]: Can it use a method reference?

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryLogging.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.jboss.logging.Logger;
+import org.jboss.logging.annotations.MessageLogger;
+
+@MessageLogger(projectCode = "SRRML")
+public interface InMemoryLogging {
+
+    InMemoryLogging log = Logger.getMessageLogger(InMemoryLogging.class, "io.smallrye.reactive.messaging");
+
+    // 19667-19999 (logging)
```
Comment (cescoffier) [inline smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryLogging.java:11]: Shouldn't it be first? It's above in the other connectors.

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryMessages.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.jboss.logging.Messages;
+import org.jboss.logging.annotations.MessageBundle;
+
+@MessageBundle(projectCode = "SRRML")
+public interface InMemoryMessages {
+
+    InMemoryMessages msg = Messages.getBundle(InMemoryMessages.class);
+
+    // 19334-19666 (messaging)
```
Comment (cescoffier) [inline smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryMessages.java:11]: Should it be before the definition? Or actually, now that I'm thinking about it, in the Javadoc of the class?

File: smallrye-reactive-messaging-jms/pom.xml [hunk 11]
```diff
@@ -113,14 +113,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-jms/pom.xml:116]: Should be kept.

File: smallrye-reactive-messaging-kafka/pom.xml [hunk 12]
```diff
@@ -73,14 +73,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/pom.xml:76]: Must be kept.

File: smallrye-reactive-messaging-mqtt-server/pom.xml [hunk 13]
```diff
@@ -85,14 +80,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt-server/pom.xml:88]: must be kept.

File: smallrye-reactive-messaging-mqtt/pom.xml [hunk 14]
```diff
@@ -30,12 +30,12 @@
       <artifactId>testcontainers</artifactId>
       <scope>test</scope>
     </dependency>
-    <dependency>
+<!--    <dependency>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt/pom.xml:33]: why is it commented?

File: smallrye-reactive-messaging-mqtt/pom.xml [hunk 15]
```diff
@@ -67,14 +67,14 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt/pom.xml:70]: must be kept.

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java [hunk 16]
```diff
@@ -41,7 +43,7 @@ public boolean isRequested() {
     @Override
     public synchronized Emitter<T> send(T msg) {
         if (msg == null) {
-            throw new IllegalArgumentException("`null` is not a valid value");
+            throw ex.illegalArgumentForValue("null");
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java:46]: do we have to pass "null" ?

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java [hunk 17]
```diff
@@ -8,11 +10,11 @@ private Validation() {
 
     public static <T> T notNull(T instance, String name) {
         if (isBlank(name)) {
-            throw new IllegalArgumentException("`name` must be set");
+            throw ex.validationForMustBeSet("name");
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java:13]: Can we have a "nameMustBeSet" and avoiding passing "name"?

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/InternalChannelRegistry.java [hunk 18]
```diff
@@ -14,51 +16,50 @@
...
     private final Map<String, List<PublisherBuilder<? extends Message<?>>>> publishers = new HashMap<>();
     private final Map<String, List<SubscriberBuilder<? extends Message<?>, Void>>> subscribers = new HashMap<>();
     private final Map<String, Emitter<?>> emitters = new HashMap<>();
 
     @Override
     public synchronized PublisherBuilder<? extends Message<?>> register(String name,
             PublisherBuilder<? extends Message<?>> stream) {
-        Objects.requireNonNull(name, NAME_MUST_BE_SET);
-        Objects.requireNonNull(stream, "'stream' must be set");
+        Objects.requireNonNull(name, msg.mustBeSet("name"));
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/InternalChannelRegistry.java:26]: Can we avoid passing "name" and have a dedicated method?

File: smallrye-reactive-messaging-vertx-eventbus/pom.xml [hunk 19]
```diff
@@ -41,14 +41,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-vertx-eventbus/pom.xml:44]: Must be kept.

File: smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java [hunk 20]
```diff
@@ -20,13 +23,13 @@
     private final long timeout;
 
     EventBusSink(Vertx vertx, VertxEventBusConnectorOutgoingConfiguration config) {
-        this.vertx = Objects.requireNonNull(vertx, "Vert.x instance must not be `null`");
+        this.vertx = Objects.requireNonNull(vertx, msg.vertXInstanceMustNotBeNull());
```
Comment (cescoffier) [inline smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java:26]: vertx, lowe-case x.

Comment (cescoffier) [reply smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java:26]: thinking about it, this check is useless. The instance cannot be `null` since the inclusion of vert.x in the core.

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java (hunk 3 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:8]: Maybe "SRMSG" instead?

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java (hunk 16 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java:46]: Maybe change the message to be about null to being valid and use a specific method name for it?

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java (hunk 3 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:8]: I am easy, just let me know what it should be.

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java (hunk 16 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java:46]: I reuse that for other fields (I think)

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java (hunk 17 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java:13]: I reuse that for other fields (I think), but I can change to have one per field

File: smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java (hunk 20 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java:26]: Should I just remove it ?

Comment (cescoffier) [reply smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java:26]: Yes, same in the source. Remove the check, but with the assignment.

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java (hunk 17 above)
Comment (cescoffier) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java:13]: it's just that "name" is often used, so we pass a constant.

**Comment by cescoffier** [comment]
Holding my current PR to avoid creating even more conflicts :-)

File: pom.xml (hunk 1 above)
Comment (phillip-kruger) [reply pom.xml:62]: Removed it from the documentation, will add something on JBoss Logging (if applicable) in the next PR when I update the examples.

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java (hunk 3 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:8]: Making it SRMSG

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryLogging.java (hunk 9 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryLogging.java:11]: I started from the bottom (for connectors). So the implementation is first, then there is a gap (for future use, either impl from the one side or new connectors from the other side. That was my thinking.

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryMessages.java (hunk 10 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryMessages.java:11]: There is a global file in the root of the parent that contains all indexes. I can move this to Javadoc, no problem :)

File: smallrye-reactive-messaging-mqtt/pom.xml (hunk 14 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-mqtt/pom.xml:33]: Sorry, removed it. Replaced with JBoss Logging

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryMessages.java (hunk 10 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/i18n/InMemoryMessages.java:11]: Moved to JavaDoc

File: smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java (hunk 20 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-vertx-eventbus/src/main/java/io/smallrye/reactive/messaging/eventbus/EventBusSink.java:26]: Removed

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java (hunk 7 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java:146]: Yup

File: smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java (hunk 8 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java:154]: done

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java (hunk 16 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/LegacyEmitterImpl.java:46]: Replace with dedicated field.

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java (hunk 17 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/helpers/Validation.java:13]: Changed to dedicated field

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/InternalChannelRegistry.java (hunk 18 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/InternalChannelRegistry.java:26]: Done, created dedicated methods

**Comment by phillip-kruger** [comment]
@kenfinnigan @cescoffier All review requests done and conflicts sorted. Please check again.
Thannks :)

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc [hunk 21]
```diff
@@ -75,19 +75,4 @@ Running the previous example should give the following output:
...
-[source,xml]
-----
-<dependency>
-  <groupId>org.slf4j</groupId>
-  <artifactId>slf4j-simple</artifactId>
-  <version>1.7.30</version>
-</dependency>
-----
-====
+----
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:78]: You should add a note but using JUL/JBoss Logging instead.

File: smallrye-reactive-messaging-cloud-events/pom.xml [hunk 22]
```diff
@@ -24,8 +24,34 @@
       <artifactId>smallrye-reactive-messaging-provider</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-cloud-events/pom.xml:27]: This module is not a connector, so no need for it.

File: smallrye-reactive-messaging-cloud-events/pom.xml [hunk 23]
```diff
@@ -24,8 +24,34 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>        
+        <configuration>
+          <generatedSourcesDirectory>${project.build.directory}/generated-sources/</generatedSourcesDirectory>
+          <annotationProcessors>
+            <annotationProcessor>
+              io.smallrye.reactive.messaging.connector.ConnectorAttributeProcessor
```
Comment (cescoffier) [inline smallrye-reactive-messaging-cloud-events/pom.xml:44]: Can be removed - not a connector

**Comment by phillip-kruger** [comment]
@cescoffier changed as requested. Please have a look :)

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc [hunk 24]
```diff
@@ -79,15 +79,18 @@ Running the previous example should give the following output:
 
 [NOTE]
 ====
-SmallRye Reactive Messaging uses SLF4J as logger.
-Check the https://www.slf4j.org/manual.html[SLF4J documentation] to configure the logs.
+SmallRye Reactive Messaging uses JBoss-logger as logger.
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:82]: JBoss logging no? 

Can you link to the project documentation page as it's not as known as SLF4J.

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc [hunk 25]
```diff
@@ -79,15 +79,18 @@ Running the previous example should give the following output:
...
 For testing purpose, you can simply add the following dependency to your `pom.xml` file:
 [source,xml]
 ----
 <dependency>
-  <groupId>org.slf4j</groupId>
-  <artifactId>slf4j-simple</artifactId>
-  <version>1.7.30</version>
+    <groupId>org.jboss.slf4j</groupId>
+    <artifactId>slf4j-jboss-logging</artifactId>
+    <version>${slf4j-jboss-logging.version}</version>
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:90]: This version should be set. The best is to use attributes for this. With Antora it's a bit messy:

* in documentation/src/main/antora/antora.yaml add a line for the JBoss logging version:
```
jboss-logging-version: '${jboss-logging.version}'
```

* in the documentation/pom.xml or in the root pom.xml declare the `jboss-logging.version` property

* in the getting-started document, use `{jboss-logging-version}` (no $) and [source,xml,subs=attributes+]

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc [hunk 26]
```diff
@@ -79,15 +79,18 @@ Running the previous example should give the following output:
...
+    <scope>test</scope>
 </dependency>
 ----
-====
+
+Check the https://www.slf4j.org/manual.html[SLF4J documentation] to configure the logs.
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:95]: I'm confused now... Do you mean that I need to configure slf4j, but I don't have any "backend" defined?

Comment (phillip-kruger) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:95]: Yea like I said I am not sure how this will work, but will update this in the next PR once I go through all examples.

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 24 above)
Comment (phillip-kruger) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:82]: I am not sure what is the correct project URL....
This (https://github.com/jboss-logging/jboss-logging) does not have any documentation.

Comment (cescoffier) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:82]: @dmlloyd Do you know where we can find the jboss-logging documentation?

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 26 above)
Comment (phillip-kruger) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc]: Ok, I removed all reference to slf4j and showed how to configure logging with JUL. I also fixed #519

File: .gitignore [hunk 27]
```diff
@@ -131,3 +131,4 @@ local.properties
 .settings
 .classpath
 .factorypath
+/examples/quickstart/nbactions.xml
```
Comment (cescoffier) [inline .gitignore:134]: netbeans?

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc [hunk 28]
```diff
@@ -26,7 +26,7 @@ Creates a Maven project, and include the following dependency in your pom.xml:
 <dependency>
   <groupId>io.smallrye.config</groupId>
   <artifactId>smallrye-config</artifactId>
-  <version>1.7.0</version>
+  <version>1.8.0</version>
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:29]: this version should be an attribute. Can you open an issue about that?

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 24 above)
Comment (cescoffier) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:82]: @phillip-kruger can you change to Jboss Logging at least?

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 28 above)
Comment (phillip-kruger) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:29]: I can just quickly fix. Please hold on.

File: .gitignore (hunk 27 above)
Comment (phillip-kruger) [reply .gitignore:134]: Yup.

Comment (phillip-kruger) [reply .gitignore:134]: I can maybe make it more general (**/nb**.xml) ?

Comment (phillip-kruger) [reply .gitignore:134]: Made it more general.

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 28 above)
Comment (phillip-kruger) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:29]: Fixed.

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 24 above)
Comment (cescoffier) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc:82]: still JBoss-logger?

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+import org.jboss.logging.Messages;
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageBundle;
+
+/**
+ * Exceptions for AMQP Connector
+ * Assigned ID range is 16000-16099
+ */
+@MessageBundle(projectCode = "SRMSG")
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:12]: It might be a good idea to add length. Though it's the same as the default right now, if the default changed we would get different numbers output

Comment (kenfinnigan) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:12]: That would apply to all `@MessageBundle` usage

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPLogging.java [hunk 30]
```diff
@@ -0,0 +1,118 @@
...
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+/**
+ * Logging for AMQP Connector
+ * Assigned ID range is 16200-16299
+ */
+@MessageLogger(projectCode = "SRMSG")
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPLogging.java:13]: Same on the length for `@MessageLogger`

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPLogging.java [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+public interface AMQPLogging {
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPLogging.java:14]: This, and other loggers, should extend `BasicLogger` as per wiki

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 24 above)
Comment (dmlloyd) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc]: No I don't.  There is some old Javadoc published at jboss.org.  Maybe @jamezp knows more?

**Comment by dmlloyd** [comment]
Comment from the peanut gallery: :peanuts: Using JUL seems a worse problem than having a dependency.

File: message-ranges.txt [hunk 32]
```diff
@@ -0,0 +1,107 @@
+# This document is intended to establish the ranges for message IDs in the various sub-projects.
+
+# Range     = owner
+
+# Provider (smallrye-reactive-messaging-provider)
+00000-00099 = exceptions
```
Comment (kenfinnigan) [inline message-ranges.txt:6]: It's odd to me that exceptions and messaging are separated, as they're the same kind of thing

File: documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc (hunk 24 above)
Comment (jamezp) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc]: There is currently no documentation for the logging facade itself. There is some documentation for the logging tooling https://jboss-logging.github.io/jboss-logging-tools/. We could publish the JavaDoc's though.

Comment (cescoffier) [reply documentation/src/main/doc/modules/ROOT/pages/getting-started.adoc]: Thanks @jamezp! That's what we were looking for. @phillip-kruger can you add a link to https://jboss-logging.github.io/jboss-logging-tools/?

**Comment by jamezp** [comment]
> Comment from the peanut gallery:  Using JUL seems a worse problem than having a dependency.

Just to add to this. If the project is already using the JBoss Logging Tooling for internationalization then JBoss Logging is already being used so it may as well be used as a facade.

File: message-ranges.txt (hunk 32 above)
Comment (phillip-kruger) [reply message-ranges.txt:6]: In this case, yes, because it's mostly use with `Objects.require...` that end up in an exception.
However one could translate something that is not in the log or an exception. However, if you want me to change it, I'll happily do so :) Just let me know.

**Comment by phillip-kruger** [comment]
@jamezp @dmlloyd I could not find any nice documentation on how to use Jboss logger (from a end user point of view, please point me in the right direction. I also don't think we should make the quick start about logging, that it why I suggested just showing how to get the log levels changed without adding more deps. But we can change it to whatever. (I just don't know how to)

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java (hunk 29 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:12]: What should the length be ? 6 ?

**Comment by jamezp** [comment]
> @jamezp @dmlloyd I could not find any nice documentation on how to use Jboss logger (from a end user point of view, please point me in the right direction. I also don't think we should make the quick start about logging, that it why I suggested just showing how to get the log levels changed without adding more deps. But we can change it to whatever. (I just don't know how to)

@phillip-kruger Sorry I was speaking specifically to the usage of `java.util.logging.Logger`'s in the code base itself, for example https://github.com/smallrye/smallrye-reactive-messaging/pull/597/files#diff-6a896a9f2624e4b85963f8a14d899547R26. 

That said JBoss Logging is just a facade so setting levels would need a log manager which could be JUL if examples don't want to add a log manager dependency.

**Comment by phillip-kruger** [comment]
@jamezp That code (the API) is code that should eventually move to MicroProfile API, so we should not introduce new dependencies there (as MP does not have any loggers in the API), so ther eI remove slf4j and did not introduced JBoss-logging as like with the rest of the modules. Hope that makes sense

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java (hunk 29 above)
Comment (phillip-kruger) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:12]: (seems like the default is 6, but the wiki says 5)

Comment (kenfinnigan) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:12]: We want the length at 5

Comment (phillip-kruger) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/i18n/AMQPExceptions.java:12]: Made it 5
